code/game/mecha/mecha_construction_paths.dm

```diff
@@ -1810,4 +1810,251 @@
+
+/datum/construction/mecha/nkarrdem_chassis/action(atom/used_atom,mob/user as mob)
+	return check_all_steps(used_atom,user)
+
+/datum/construction/mecha/nkarrdem_chassis/spawn_result(mob/user, result_name)
+	..(user, "Nkarrdem")
+	var/obj/item/mecha_parts/chassis/const_holder = holder
+	const_holder.construct = new /datum/construction/reversible/mecha/nkarrdem(const_holder)
+	const_holder.icon = 'icons/mecha/mech_construction.dmi'
+	const_holder.icon_state = "nkarrdem0"
+	const_holder.density = TRUE
+	qdel(src)
+	return
+
+/datum/construction/reversible/mecha/nkarrdem
+	result = "/obj/mecha/janitor/nkarrdem"
+	steps = list(
+		//1
+		list(
+			"key" = TOOL_WELDER,
+			"backkey" = TOOL_WRENCH,
+			"desc" = "External armor is wrenched."),
+		//2
+		list(
+			"key" = TOOL_WRENCH,
+			"backkey" = TOOL_CROWBAR,
+			"desc" = "External armor is installed."),
+		//3
+		list(
+			"key" = /obj/item/stack/sheet/plasteel,
+			"backkey" = TOOL_WELDER,
+			"desc" = "Internal armor is welded."),
+		//4
+		list(
+			"key" = TOOL_WELDER,
+			"backkey" = TOOL_WRENCH,
+			"desc" = "Internal armor is wrenched."),
+		//5
+		list(
+			"key" = TOOL_WRENCH,
+			"backkey" = TOOL_CROWBAR,
+			"desc" = "Internal armor is installed."),
+		//6
+		list(
+			"key" = /obj/item/stack/sheet/metal,
+			"backkey" = TOOL_SCREWDRIVER,
+			"desc" = "Floor buffer is secured."),
+		//7
+		list(
+			"key" = TOOL_SCREWDRIVER,
+			"backkey" = TOOL_CROWBAR,
+			"desc" = "Floor buffer is installed."),
+		//8
+		list(
+			"key" = /obj/item/borg/upgrade/floorbuffer,
+			"backkey" = TOOL_SCREWDRIVER,
+			"desc" = "Peripherals control module is secured."),
+		//9
+		list(
+			"key" = TOOL_SCREWDRIVER,
+			"backkey" = TOOL_CROWBAR,
+			"desc" = "Peripherals control module is installed."),
+		//10
+		list(
+			"key" = /obj/item/circuitboard/mecha/nkarrdem/peripherals,
+			"backkey" = TOOL_SCREWDRIVER,
+			"desc" = "Central control module is secured."),
+
+		//11
+		list(
+			"key" = TOOL_SCREWDRIVER,
+			"backkey" = TOOL_CROWBAR,
+			"desc" = "Central control module is installed."),
+		//12
+		list(
+			"key" = /obj/item/circuitboard/mecha/nkarrdem/main,
+			"backkey" = TOOL_SCREWDRIVER,
+			"desc" = "The wiring is adjusted."),
+		//13
+		list(
+			"key" = /obj/item/wirecutters,
+			"backkey" = TOOL_SCREWDRIVER,
+			"desc" = "The wiring is added."),
+		//14
+		list(
+			"key" = /obj/item/stack/cable_coil,
+			"backkey" = TOOL_SCREWDRIVER,
+			"desc" = "The hydraulic systems are active."),
+		//15
+		list(
+			"key" = TOOL_SCREWDRIVER,
+			"backkey" = TOOL_WRENCH,
+			"desc" = "The hydraulic systems are connected."),
+		//16
+		list(
+			"key" = TOOL_WRENCH,
+			"desc" = "The hydraulic systems are disconnected.")
+	)
+
+/datum/construction/reversible/mecha/nkarrdem/action(atom/used_atom,mob/user as mob)
+	return check_step(used_atom,user)
+
+/datum/construction/reversible/mecha/nkarrdem/custom_action(index, diff, atom/used_atom, mob/user)
+	if(!..())
+		return 0
+
+	//TODO: better messages.
```

Comment:
Is this still relevant?

Reply:
No, this comment is a leftover. Removed.

The `if(!..())` handles the stuff from `mecha/custom_action` which is used to handle material counts and tools, so it should remain.

Reply:
I meant the comment. I should have been more clear.